source/admin_manual/list_of_events.rst

```diff
@@ -2726,6 +2726,91 @@ Event emitted when dict server finishes transaction. Same fields as
+| Field               | Description                                          |
++=====================+======================================================+
+| Inherits from :ref:`pre_login_client`                                      |
++---------------------+------------------------------------------------------+
+| service             | Name of service e.g. ``submission``, ``imap``        |
++---------------------+------------------------------------------------------+
+| dest_host           | Host name of the proxy remote (if proxying is        |
+|                     | configured with IP address, will have the same value |
+|                     | as dest_ip).                                         |
++---------------------+------------------------------------------------------+
+| dest_ip             | Destination IP for proxying                          |
++---------------------+------------------------------------------------------+
+| dest_port           | Destination port for proxying                        |
++---------------------+------------------------------------------------------+
+| source_ip           | Source IP where proxying originated from             |
++---------------------+------------------------------------------------------+
+
+login_proxy_started
+^^^^^^^^^^^^^^^^^^^
+Emitted before connecting to proxy remote.
+
+login_proxy_logged_in
+^^^^^^^^^^^^^^^^^^^^^
+Emitted after proxied connection is established and user is successfully logged
+in to the backend.
+
++---------------------+------------------------------------------------------+
+| Field               | Description                                          |
++=====================+======================================================+
+| source_port         | Source port where proxying originated from (present  |
+|                     | after proxy connection is established                |
++---------------------+------------------------------------------------------+
+| master_user         | If proxying is done with a master user               |
+|                     | authentication, contains the full username of master |
+|                     | user.                                                |
++---------------------+------------------------------------------------------+
+
+login_proxy_finished
+^^^^^^^^^^^^^^^^^^^^
+Emitted when proxying has ended. Either successfully or with error.
+
++-----------------+------------------------------------------------------+
+| Field           | Description                                          |
++=================+======================================================+
+| source_port     | Source port where proxying originated from (present  |
+|                 | after proxy connection is established                |
++-----------------+------------------------------------------------------+
+| error           | For failed proxy connections, contains the error.    |
```

Comment:
I think this can only happen when login_proxy_logged_in event hasn't been sent? Could clarify a bit then, so it's not sent if e.g. client or server unexpectedly disconnects after login has been successful.

Reply:
Reworded a bit. Hopefully it's clearer now.